File: src/main/java/ru/translater/format/FormatProperties.java
package ru.translater.format;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;
import ru.translater.common.CommonProperties;

@Data
@Configuration
@ConfigurationProperties(prefix = "format")
public class FormatProperties extends CommonProperties {
    private String analyzePath;
    private String formatTemplatePath;
}

File: src/main/kotlin/com/translater/format/properties/FormatProperties.kt
package com.translater.format.properties

import com.translater.common.properties.CommonProperties
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConstructorBinding

@ConstructorBinding
@ConfigurationProperties(prefix = "format")
class FormatProperties(
    override val storePath: String,
    val analyzePath: String,
    val formatTemplatePath: String
) : CommonProperties(storePath)